pkg/hubble/parser/common/endpoint.go

```diff
@@ -75,41 +74,36 @@ func (r *EndpointResolver) ResolveEndpoint(ip netip.Addr, datapathSecurityIdenti
 				Labels:    SortAndFilterLabels(r.log, ep.GetLabels(), identity.NumericIdentity(epIdentity)),
 				PodName:   ep.GetK8sPodName(),
 			}
-			if pod := ep.GetPod(); pod != nil {
-				workload, workloadTypeMeta, ok := utils.GetWorkloadMetaFromPod(pod)
-				if ok {
-					e.Workloads = []*pb.Workload{{Kind: workloadTypeMeta.Kind, Name: workload.Name}}
-				}
+			if workload := ep.GetWorkload(); workload != nil {
+				e.Workloads = []*pb.Workload{{Kind: workload.Kind, Name: workload.Name}}
 			}
 			return e
 		}
 	}
 
 	// for remote endpoints, assemble the information via ip and identity
-	numericIdentity := datapathSecurityIdentity
-	var namespace, podName string
+	e := pb.Endpoint{
```

Comment:
Is there a reason to move this up here and assigning to the fields, rather than the old method which just returned the `pb.Endpoint` with each field set? I feel like this approach makes it easier to miss setting fields, since it's broken across multiple conditions and lines.

Reply:
Taking performance into account, as I minimize variable creation. If you consider the improvement to be marginal and not worth sacrificing code readability, I can revert back to the original approach. :-)